File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(arr) {
       // no array return 0
    if(!arr.length) return 0;

   let maxProfitResult = 0;
   
  let minPrice = arr[0];
   
 //lopp tough the array 
 for(let i =1 ; i < arr.length; i++){
      let maxRunningP = arr[i] - minPrice;

   maxProfitResult = Math.max(maxProfitResult, maxRunningP);
   minPrice = Math.min(minPrice, arr[i])

 }


return maxProfitResult

};

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        buy = prices[0]
        
        profit = 0
        
        for i in range(len(prices)):
            
            if prices[i] - buy >0:
                print(prices[i]-buy)
                profit = max(profit,prices[i]- buy)
            
            buy = min(buy, prices[i])
            
        return profit  
        